code/__HELPERS/game.dm

```diff
@@ -213,26 +213,189 @@
+//and each is given references to the movables they represent. that way you can do for(var/mob/oranges_ear/ear in view(...)) and check what they reference
+//as opposed to for(var/atom/movable/target in view(...)) and checking if they have the properties you want which leads to much larger lists generated by view()
+//and also leads to iterating through more movables to filter them.
+//TLDR: iterating through just mobs is much faster than all movables when iterating through view(), this system leverages that to boost speed
+
+//named because the idea was first made by oranges and i didnt know what else to call it (note that this system was originally made for get_hearers_in_view())
+/mob/oranges_ear
+	icon_state = null
+	density = FALSE
+	move_resist = INFINITY
+	invisibility = 0
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	logging = null
+	held_items = null //all of these are list objects that should not exist for something like us
+	faction = null
+	alerts = null
+	screens = null
+	client_colours = null
+	hud_possible = null
+	/// references to everything "on" the turf we are assigned to, that we care about. populated in assign() and cleared in unassign()
+	/// movables iside of other movables count as being "on" if they have get_turf(them) == our turf. intentionally not a lazylist
+	var/list/references = list()
+
+/mob/oranges_ear/Initialize(mapload)
+	SHOULD_CALL_PARENT(FALSE)
+	return INITIALIZE_HINT_NORMAL
+
+/mob/oranges_ear/Move()
+	SHOULD_CALL_PARENT(FALSE)
+	stack_trace("SOMEHOW A /mob/oranges_ear MOVED")
+	return FALSE
+
+/mob/oranges_ear/abstract_move(atom/destination)
+	SHOULD_CALL_PARENT(FALSE)
+	stack_trace("SOMEHOW A /mob/oranges_ear MOVED")
+	return FALSE
+
+/mob/oranges_ear/Bump()
+	SHOULD_CALL_PARENT(FALSE)
+	return FALSE
+
+/**
+ * assigns this ear to the turf at get_turf(assigned_atom) and adds it to this ears references list then moves to the turf.
+ * used when SSspatial_grid assigns oranges_ear instances to a list of movables such that there will only be one oranges_ear per turf
+ */
+/mob/oranges_ear/proc/assign(atom/assigned_atom)
+	var/turf/assigned_turf_loc = get_turf(assigned_atom)
+	if(!assigned_turf_loc)
+		return FALSE
+
+	if(assigned_turf_loc.assigned_oranges_ear)
+		assigned_turf_loc.assigned_oranges_ear.references |= assigned_atom
+		return FALSE//we dont put ourselves in assigned_turf_loc if it already has an oranges ear, we just give it assigned_atom and dont move there
+
+	references += assigned_atom
+
+	loc = assigned_turf_loc//normally setting loc directly like this isnt allowed but we literally JUST need to exist in that loc without wasting cpu time
+	assigned_turf_loc.assigned_oranges_ear = src
+
+	return TRUE
+
+///clean this oranges_ear up for future use
+/mob/oranges_ear/proc/unassign()
+	loc:assigned_oranges_ear = null//trollface. our loc should ALWAYS be a turf, no exceptions. if it isnt then this doubles as an error message ;)
```

Comment:
If you do this I can't bully goon over having a : in their flow code
Is this really so hot you can't afford a typecast, even with no check